todo_app/app.py

```diff
@@ -1,14 +1,55 @@
+import os
+from urllib.parse import urlencode
 from flask import Flask, render_template, request, redirect
+from flask_login import LoginManager, login_required, login_user
+import requests
 from todo_app.data import mongo_items
 from todo_app.flask_config import Config
 from todo_app.view_models.ViewModel import ViewModel
+from todo_app.authentication.user import User
 
 def create_app(): 
 
     app = Flask(__name__)
     app.config.from_object(Config())
 
+    app.config['LOGIN_DISABLED'] = os.getenv('LOGIN_DISABLED') == 'True'
+
+    login_manager = LoginManager()
+
+    @login_manager.unauthorized_handler
+    def unauthenticated():
+        params = {
+            'client_id': os.getenv('OAUTH_CLIENT_ID')
+        }
+
+        encoded_params = urlencode(params)
+    
+        return redirect(f'https://github.com/login/oauth/authorize?{encoded_params}')
+       
+
+    @login_manager.user_loader
+    def load_user(user_id):
+        app_user = User(id=user_id)
+        return app_user
+        
+
+    login_manager.init_app(app)
+
+    @app.route('/login/callback')
+    def callback():
+        access_token = __get_access_token()
```

Comment:
Interesting use of the double_leading_underscore approach - in some hundred+ reviews of python exercises I don't think I've seen this used and I [learned something new today about name mangling](https://realpython.com/python-double-underscore/#double-leading-underscore-in-classes-pythons-name-mangling)! 😄

I guess my follow up comment is that from poking around it sounds like this is a fairly rarely used language feature with most just using the single underscore approach so nothing wrong with it but worth being aware it's a rarer pattern!